### PR TITLE
[1] fix: remove quote API usage for fiat amount and price impact calc

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

In this PR I just removed Quote API usage form everywhere besides except for displaying the price.

1. Removed Quote API usage for fiat amounts (https://github.com/cowprotocol/cowswap/pull/3074/commits/0436ad0b5b7f29ab3551fddce6cb295d74bf3479)
2. Moved `useHigherUSDValue` hook to a new module `modules/fiatAmount`. This hook is commonly used to get a fiat amount (https://github.com/cowprotocol/cowswap/pull/3074/commits/76925b3a3e16a62ba79a0dc8ae4c948bc254c449)
3. Removed Quote API usage for price impact calculations (https://github.com/cowprotocol/cowswap/pull/3074/commits/3ff2993b2124c9c8a1074b9f6561c80678f1c2d6)

  # To Test

Please, do not test this PR. This one is an intermediate PR in a waterfall.
The final PR: https://github.com/cowprotocol/cowswap/pull/3081

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/trade/hooks/useTradePriceImpactParams.ts [hunk 1]
```diff
@@ -10,29 +10,15 @@ const emptyState = {
   outputCurrencyAmount: null,
 }
 
-export function useTradePriceImpactParams(): PriceImpactParams {
+export function useTradePriceImpactParams(): ParsedAmounts {
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/trade/hooks/useTradePriceImpactParams.ts:13]: This hook is now `useParsedAmounts`, no?

File: apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        quoteAmount,
+      }).invert()
+
+      console.debug(
+        '[useCoingeckoUsdPrice] Best Coingecko USD price amount',
+        usdPrice.toSignificant(12),
+        usdPrice.invert().toSignificant(12)
+      )
+
+      return setPrice(usdPrice)
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:104]: Why returning the `setPrice`?

File: apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    } catch (error: any) {
+      console.error(
+        '[useStablecoinPrice::useCoingeckoUsdPrice]::Error getting USD price from Coingecko for token',
+        tokenAddress,
+        error
+      )
+      return batchedUpdate(() => {
+        setError(new Error(error))
+        setPrice(null)
+      })
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:114]: I thought react hooks were smart enough at this point to no longer need batching. Not the case?

File: apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  value: CurrencyAmount<Token> | null
+  isLoading: boolean
+} {
+  return useMemo(() => {
+    if (!price || error || !currencyAmount) return { value: null, isLoading }
+
+    try {
+      return { value: price.invert().quote(currencyAmount), isLoading }
+    } catch (error: any) {
+      return { value: null, isLoading }
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:36]: Why not return the error?

Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:36]: I actually didn't change this code.
Sorry for the noise.
This file is a moved [useStablecoinPrice/index.ts](https://github.com/cowprotocol/cowswap/pull/3074/files#diff-931a80cf03895977369e7dba4f68ee9d31844632a1e13c8f57a0c0bd50f1ab4e)

File: apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:104]: I actually didn't change this code.
Sorry for the noise.
This file is a moved [useStablecoinPrice/index.ts](https://github.com/cowprotocol/cowswap/pull/3074/files#diff-931a80cf03895977369e7dba4f68ee9d31844632a1e13c8f57a0c0bd50f1ab4e)

File: apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts (hunk 3 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:114]: I actually didn't change this code.
Sorry for the noise.
This file is a moved [useStablecoinPrice/index.ts](https://github.com/cowprotocol/cowswap/pull/3074/files#diff-931a80cf03895977369e7dba4f68ee9d31844632a1e13c8f57a0c0bd50f1ab4e)

File: apps/cowswap-frontend/src/modules/swap/containers/TradeSummary/index.tsx [hunk 5]
```diff
@@ -1,11 +1,12 @@
 import { Percent } from '@uniswap/sdk-core'
 
-import { useHigherUSDValue } from 'legacy/hooks/useStablecoinPrice'
 import TradeGp from 'legacy/state/swap/TradeGp'
 
 import { TradeSummaryContent } from 'modules/swap/pure/TradeSummary'
 import { useWalletDetails } from 'modules/wallet'
 
+import { useHigherUSDValue } from '../../../fiatAmount/hooks/useHigherUSDValue'
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/swap/containers/TradeSummary/index.tsx]: Thanks!

File: apps/cowswap-frontend/src/modules/trade/hooks/useTradePriceImpactParams.ts (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/trade/hooks/useTradePriceImpactParams.ts:13]: Renamed, thanks!

File: apps/cowswap-frontend/src/modules/trade/updaters/PriceImpactUpdater.tsx [hunk 6]
```diff
@@ -10,11 +10,11 @@ import { priceImpactAtom } from '../state/priceImpactAtom'
 export function PriceImpactUpdater() {
   const updatePriceImpact = useSetAtom(priceImpactAtom)
   const params = useTradePriceImpactParams()
-  const { error, loading, priceImpact } = usePriceImpact(params)
+  const { isLoading, priceImpact } = useFiatValuePriceImpact(params)
 
   useSafeEffect(() => {
-    updatePriceImpact({ error, loading, priceImpact })
-  }, [error, loading, updatePriceImpact, priceImpact])
+    updatePriceImpact({ error: undefined, loading: isLoading, priceImpact })
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/trade/updaters/PriceImpactUpdater.tsx]: Removed it, thanks!

File: apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts (hunk 4 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts:36]: Also, this file is deleted in the next PR

File: apps/cowswap-frontend/src/legacy/state/swap/hooks.tsx [hunk 7]
```diff
@@ -225,19 +224,13 @@ export function useUnknownImpactWarning(priceImpactParams?: PriceImpact) {
...
   impactWarningAccepted: boolean
   priceImpactParams?: PriceImpact
   isExpertMode: boolean
 }) {
   if (isExpertMode || impactWarningAccepted) return true
-  else {
-    if (priceImpactParams?.error) {
-      return impactWarningAccepted
-    }
-  }
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/state/swap/hooks.tsx:240]: Oh, don't we need this warning?
Is it being captured somewhere else?

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/legacy/state/swap/hooks.tsx (hunk 7 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/state/swap/hooks.tsx:240]: You saved me, thanks!
Fixed it in the next PR:
https://github.com/cowprotocol/cowswap/pull/3081/commits/d0019d6f6c7ed3f09f77ab4126abf6d4e0c46011

**Review by anxolin — COMMENTED** [review]
Sasha, but are we removing ABA? 
Shouldn't we trigger always ABA as a last resource for price impact?

**Comment by shoom3301** [comment]
>Shouldn't we trigger always ABA as a last resource for price impact?

Probably yes, but with a new implementation. The previous one called ABA calculation undoubtedly and it made excessive quote calls

**Pull request merged**